### PR TITLE
fwupd: 1.8.10 -> 1.8.12

### PR DESCRIPTION
**Pull request by r-vdp**
###### Description of changes

Changelog: https://github.com/fwupd/fwupd/releases/tag/1.8.12

Added pandoc as a native build dependency because it is now required to generate the manpages (see fwupd/fwupd#5566).

Separate commit to reformat the file using nixpkgs-fmt.

@jtojnar I disabled the new device-tests that require network access to download emulation data to emulate different devices. I think that these tests are more useful for the fwupd CI than for our NixOS CI, but I'd like to hear your opinion on this.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/firmware/fwupd/default.nix [hunk 1]
```diff
@@ -265,6 +271,14 @@ let
...
       # in nixos test tries to chmod 0777 $out/share/installed-tests/fwupd/tests/redfish.conf
       sed -i "s/get_option('tests')/false/" plugins/redfish/meson.build
+
+      # Device tests use device emulation and need to download emulation data from
+      # the internet, which does not work on our test VMs.
+      # It's probably better to disable these tests for NixOS by setting
+      # the device-tests directory to /dev/null.
+      # For more info on device emulation, see:
+      #   https://github.com/fwupd/fwupd/blob/eeeac4e9ba8a6513428b456a551bffd95d533e50/docs/device-emulation.md
+      sed -i "s:join_paths(datadir, 'fwupd', 'device-tests'):'/dev/null':" data/installed-tests/meson.build
```
Comment (R-VdP) [inline pkgs/os-specific/linux/firmware/fwupd/default.nix:281]: @jtojnar device tests are disabled here, by replacing the directory containing the test by `/dev/null`.

Comment (jtojnar) [reply pkgs/os-specific/linux/firmware/fwupd/default.nix:281]: Please use `substituteInPlace` unless a regular expression is needed. Unlike `sed`, that will notify us when the replacement no longer works.

File: pkgs/os-specific/linux/firmware/fwupd/default.nix [hunk 2]
```diff
@@ -79,10 +80,11 @@ let
...
   haveFlashrom = isx86 && enableFlashrom;
 
-  runPythonCommand = name: buildCommandPython: runCommand name {
-    nativeBuildInputs = [ python3 ];
+  runPythonCommand = name: buildCommandPython: runCommand name
+    {
+      nativeBuildInputs = [ python3 ];
       inherit buildCommandPython;
-  } ''
+    } ''
```
[the remark this answers is not in the document]
Comment (R-VdP) [reply pkgs/os-specific/linux/firmware/fwupd/default.nix]: Right, that seems to be a big in nixpkgs-fmt. I put the string on a single line to work around that for now.

File: pkgs/os-specific/linux/firmware/fwupd/default.nix [hunk 3]
```diff
@@ -396,4 +414,5 @@ let
     };
   };
 
-in self
+in
+self
```
[the remark this answers is not in the document]
Comment (R-VdP) [reply pkgs/os-specific/linux/firmware/fwupd/default.nix]: Good point, done!

**Comment by R-VdP** [comment]
> Although the tests are supposed to be skipped when the data is not available according to https://github.com/fwupd/fwupd/blob/eeeac4e9ba8a6513428b456a551bffd95d533e50/docs/device-emulation.md
>
> Not sure how relevant they are for us. Will defer to you.

@jtojnar the tests seem to download the missing data from the fwupd site, which is why they fail to run in the nix sandbox. I don't think it's worth it to create FODs to prefetch them.

**Review by jtojnar — APPROVED** [review]
Thanks. I moved the commits around a bit for cleaner history.

File: pkgs/os-specific/linux/firmware/fwupd/default.nix [hunk 4]
```diff
@@ -103,266 +111,282 @@ let
...
-    outputs = [ "out" "lib" "dev" "devdoc" "man" "installedTests" ];
-
-    src = fetchurl {
-      url = "https://people.freedesktop.org/~hughsient/releases/fwupd-${version}.tar.xz";
-      hash = "sha256-vvNUidNdhW9xeksjEVnkIR7CZ4oBQizZJRMFtZUq6Ow=";
     };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "fwupd";
+  version = "1.8.10";
```
Comment (K900) [inline pkgs/os-specific/linux/firmware/fwupd/default.nix:127]: Forgot to update this, now it's all broken.

**Pull request merged**